imports/api/detectors/methods.js

```diff
@@ -0,0 +1,72 @@
+import { Meteor } from 'meteor/meteor';
+import { ValidatedMethod } from 'meteor/mdg:validated-method';
+import { SimpleSchema } from 'meteor/aldeed:simple-schema';
+import { _ } from 'meteor/underscore';
+
+
+export const matchAffordancesWithDetector = new ValidatedMethod({
+  name: 'detectors.matchAffordancesWithDetector',
+
+  validate: new SimpleSchema({
+    userId: { type: String },
+    detectorId: { type: String }
+  }).validator(),
+
+  run({ userId, detectorId }) {
+    const userloc = Locations.findOne({ uid: userId });
+    const detector = Detectors.findOne({ _id: detectorId });
+
+    if (_.isEmpty(userloc.affordances)) {
+      throw new Meteor.Error('detectors.matchAffordancesWithDetector',
+        'Cannot match when a user affordances have not been sensed')
+    }
+
+    detector_output = applyDetector(userloc.affordances,
+                                    detector.variables,
+                                    detector.rules);
+    console.log('detector_output')
+    console.log(detector_output)
+
+    return detector_output;
+  }
+});
+// Test Method in Client Like So:
+// Meteor.call('detectors.matchAffordancesWithDetector', {
+//   userId: 'GozT7ZtXuu3diPPWc',
+//   detectorId: 'cG4RJxsodeYfG7ATT'
+// }, (err, res) => {
+//   if (err) {
+//     alert(err);
+//   } else {
+//     console.log(res);
+//   }
+// });
+
+/**
+ * Evaluates given the affordances of a user, if they match the definition given
```

Comment:
i'm confused by what this function does? what does matching the definition given by the detector mean? when would the affordances of a user not match the definition given by the detector? The code/function name looks like it should return true/false depending if the user matches the situation or not?

Reply:
See my changes, as I tried to rename variable names to make this more transparent.